Issue 2799 - link tags should self close

emzyme20 opened the pull request:
- The TagBuilder needs to be created with the RenderMode set to SelfClosing.
- To allow this to pass through to the ResourceDefinition via the settings, we need a new RequireSetting property called SelfClosing.

Fixes #2799

dnfclas commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=2817) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=2817) before we can accept your contribution.<br/><br/>:x: emzyme20 [sign now](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=2817)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/OrchardCMS/OrchardCore?pullRequest=2817) it.</sub>

sebastienros commented:
Should all links and meta be self closing, and no settings would be required?

emzyme20 commented:
> Should all links and meta be self closing, and no settings would be required?

I have reworked my changes.. I didn't notice until investigating your question that you already have predetermined TagRenderMode dictionary defined. I have added stylesheet to that and then you just need to set it when you create the TagBuilder - works a treat.. 

Have a look and see what you think now...
